File: src/etch/bindings/csharp/transport/DataSource.cs
using System;
using System.Text;
using System.Collections.Generic;
using Etch.Support;

namespace Etch.Transport
{
    /// <summary>
    /// Interface used to model a stream source to a data recipient, which
    /// allows the recipient to send data to the peer.
    /// </summary>
    public interface DataSource : Source 
    {

        /// <summary>
        /// Delivers data to the peer via the data stream.
        /// </summary>
        /// <param name="recipient"></param>
        /// <param name="buf"></param>
        /// Exception:
        ///     throws Exception
        void Data( Who recipient, FlexBuffer buf );
    }
}

File: binding-ruby/src/main/ruby/transport/data_source.rb
require 'etch/bindings/ruby/transport/source.rb'

module DataSource 
      include Source
  
  # Interface used to model a stream source to a data recipient, which
  # allows the recipient to send data to the peer.
  #
  #  Delivers data to the peer via the data stream.
  # @param recipient
  # @param buf
  # @throws Exception
  #
  def data( recipient, buf ) 
    raise "Subclass responsibility"
  end
   
end

